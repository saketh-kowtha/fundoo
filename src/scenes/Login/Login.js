/**
 * @author Kowtha Saketh
 * @description Login Scene
 */
import React from 'react'
import { Card, Link, Button, LanguageSelection, List } from '../../components'
import Input from './controllers/Input/Input'
import {
    validateUsername,
    validatePassword,
    validateEmail,
    validateLogin,
    validateSignUp
} from './controllers/Input/InputValidation'
import './Login.scss'
import geti18N from '../../strings'
const {
    title,
    firstName,
    lastName,
    email,
    password,
    confirmPassword,
    signIn,
    signUp,
    signUpDesc,
    signInDesc,
    forgotPasswd,
    createAccount,
    userName,
    send,
    forgetPasswordDesc,
    findYourMail
} = geti18N()
import http from '../../services/http'
import showToast from '../../components/Toast'
import { LOGIN, RESETPASSWORD, ERROR, SIGNUP } from '../../constants'
import { withRouter } from "react-router-dom";
import { connect } from 'react-redux';
import {addUser} from '../../actions/userActions'
/**
 * @name Login
 */
class Login extends React.Component {
    //Initlizing State
    state = {
        login: {
            username: {
                value: "",
                errMsg: ""
            },
            password: {
                value: "",
                errMsg: ""
            }
        },
        reset: {
            email: {
                value: "",
                errMsg: ""
            }
        },
        signUp: {
            firstname: {
                value: "Afs",
                errMsg: ""
            },
            lastname: {
                value: "Als",
                errMsg: ""
            },
            password: {
                value: "Admin@123",
                errMsg: ""
            },
            confirmpassword: {
                value: "Admin@123",
                errMsg: ""
            },
            email: {
                value: "aadarsh@gmail.com",
                errMsg: ""
            }
        },
        state: this.props.view || LOGIN,
        isSuggestionList: true
    }


    componentDidMount = () => {
        this._isMount = true
        let usersList = localStorage.getItem("usersList")
        if (!usersList)
            this.setState({ isSuggestionList: false })
    }

    componentWillUnmount = () => {
        this._isMount = false
    }

    /**
     * @name updateState
     * @description This method used like state middleware in this page
     */
    updateState = (feild, value) => {
        if (!this._isMount)
            return
        if (!feild)
            this.setState({ value })
        this.setState({ feild: value })
    }

    //User input handling
    handleUserInput = (name, input) => {
        const stateData = { ...this.state }
        stateData[this.state.state][name].value = input
        this.updateState(null, stateData)
    }


    //Signup API response handler
    signupResponseHandler = (err, data) => {
        if (err) {
            const stateData = { ...this.state.signUp }
            Object.keys(err).forEach(key => {
                let errMsg = (Array.isArray(err[key]) ? err[key].join(",") : err[key])
                stateData[key].errMsg = errMsg
                showToast(errMsg, ERROR)
            })
            this.updateState([this.state.state], stateData)
        }
        else
            showToast(data)
    }

    //Reset Password API response handler
    resetResponseHandler = (err, data) => {
        if (err)
            showToast(err, ERROR)
        else
            showToast(data)
    }

    //Login API response handler
    loginResponseHandler = (err, data) => {
        if (err) {
            showToast(err, ERROR)
        }
        else if(data){
            let user = localStorage.getItem("usersList")

            if (user) {
                user = user.split(",")
                user.push(data.email)
            }
            else
                user = [data.email]
            localStorage.setItem("usersList", user.filter((userName, index) => user.indexOf(userName) == index).join(",") )
            this.props.addUser(data)
            this.props.history.push("/Dashboard")
        }
    }

    //This metohod will remove all error before validation
    clearErrorMsgs = () => {
        let stateData = { ...this.state[this.state.state] }
        Object.keys(stateData).forEach(feild => {
            stateData[feild].errMsg = ""
        })
        this.updateState([this.state.state], stateData)
    }

    //Common validation method for signup, login and reset
    validate = () => {
        this.clearErrorMsgs()
        if (this.state.state === LOGIN) {
            const stateData = { ...this.state.login }
            const loginValidation = validateLogin(this.state.login)
            if (loginValidation != true) {
                stateData[loginValidation.feild].errMsg = loginValidation.errorMsg
                // console.log(loginValidation, stateData[loginValidation.feild])
                this.updateState("login", stateData)
                // this.setState({login: stateData})
            }
            else if (loginValidation === true) {
                http.login(stateData, this.loginResponseHandler)
            }
        }
        else if (this.state.state === RESETPASSWORD) {
            const resetValidation = validateEmail(this.state.reset.email.value)
            const stateData = { ...this.state.reset }
            if (resetValidation != true) {
                stateData['email'].errMsg = resetValidation
                // this.setState({reset: stateData})    
                this.updateState("reset", stateData)
            }
            else
                http.forgotPassword(stateData, this.resetResponseHandler)
        }
        else {
            const signUpValidation = validateSignUp(this.state.signUp)
            const stateData = { ...this.state.signUp }
            if (signUpValidation != true) {
                stateData[signUpValidation.feild].errMsg = signUpValidation.errorMsg
                // this.setState({signUp: stateData})
                this.updateState("signUp", stateData)

            }
            else if (signUpValidation === true) {
                http.signUp(stateData, this.signupResponseHandler)
            }
        }
    }

    //Forget password JSX component
    ForgetPassword = () => {
        return <div>
            <div className="header">
                <p>{title.split("").map((char, index) => <span key={index}>{char}</span>)}</p>
                <h2>{findYourMail}</h2>
                <p>{forgetPasswordDesc}</p>
            </div>
            <div>
                <Input
                    name={"email"}
                    key={email}
                    label={email}
                    value={this.state.reset.email.value}
                    errorMsg={this.state.reset.email.errMsg}
                    onChange={this.handleUserInput}
                    validate={validateEmail} />
            </div>
            <div className="footer">
                <div>
                    <Link
                        name={signIn}
                        small={"true"}
                        onClick={() => this.props.history.push("/login")} />
                </div>
                <div>
                    <Button onClick={this.validate}>{send}</Button>
                </div>
            </div>
        </div>
    }


    setUsername = (name) => {
        if(name === "Other Account")
            name = ""
        const stateData = { ...this.state }
        stateData.isSuggestionList = false
        stateData.login.username.value = name
        this.updateState(null, stateData)
        this.setState({ isSuggestionList: false })
    }

    //LoginUserSuggestionJSX component
    LoginUserSuggestion = () => {
        let usersList = localStorage.getItem("usersList")
        if (!usersList)
            return
        usersList = usersList.split(",").map(element => ({label: element.trim(), icon: 'perm_identity'}))
        usersList.push({label: "Other Account", icon: 'perm_identity'})
        return <div>
             <List className="login-users-list" data={usersList} onSelect={(e) => this.setUsername(e)}/>
        </div>
    }

    //Signup JJSX component
    SignUp = () => {
        let _this = { ...this.state.signUp }
        return <div>
            <div className="header">
                <p>{title.split("").map((char, index) => <span key={index}>{char}</span>)}</p>
                <h2>{signUp}</h2>
                <p>{signUpDesc}</p>
            </div>
            <div className="row">
                <Input
                    name={"firstname"}
                    key={firstName}
                    label={firstName}
                    value={_this.firstname.value}
                    errorMsg={_this.firstname.errMsg}
                    onChange={this.handleUserInput}
                    validate={validateUsername} />

                <Input
                    name={"lastname"}
                    key={lastName}
                    label={lastName}
                    value={_this.lastname.value}
                    errorMsg={_this.lastname.errMsg}
                    onChange={this.handleUserInput}
                    validate={validateUsername} />
            </div>
            <div className="row" >
                <Input
                    name={"email"}
                    key={email}
                    label={email}
                    value={_this.email.value}
                    errorMsg={_this.email.errMsg}
                    onChange={this.handleUserInput}
                    validate={validateEmail} />
            </div>
            <div className="row">
                <Input
                    name={"password"}
                    key={password}
                    label={password}
                    value={_this.password.value}
                    errorMsg={_this.password.errMsg}
                    secret={true}
                    onChange={this.handleUserInput}
                    validate={validatePassword} />
                <Input
                    name={"confirmPassword"}
                    key={confirmPassword}
                    label={confirmPassword}
                    value={_this.confirmpassword.value}
                    errorMsg={_this.confirmpassword.errMsg}
                    onChange={this.handleUserInput}
                    secret={true}
                    validate={validatePassword} />
            </div>
            <div className="footer">
                <div>
                    <Link
                        name={signIn}
                        small={"true"}
                        onClick={() => this.props.history.push("/login")} />
                </div>
                <div>
                    <Button
                        onClick={this.validate}>
                        {signUp}
                    </Button>
                </div>
            </div>
        </div>
    }

    //Login component
    Login = () => {
        return <div>
            <div className="header">
                <p>{title.split("").map((char, index) => <span key={index}>{char}</span>)}</p>
                <h2>{signIn}</h2>
                <p>{signInDesc} </p>
            </div>
            <div>
                <Input
                    name={"username"}
                    key={userName}
                    label={userName}
                    value={this.state.login.username.value}
                    errorMsg={this.state.login.username.errMsg}
                    onChange={this.handleUserInput}
                    validate={validateEmail} />
                <Input
                    name={"password"}
                    key={password}
                    label={password}
                    errorMsg={this.state.login.password.errMsg}
                    onChange={this.handleUserInput}
                    secret={true}
                    validate={validatePassword} />
            </div>
            <div>
                <Link name={forgotPasswd} small={"true"} onClick={() => this.props.history.push("/resetpassword")} />
            </div>
            <div className="footer">
                <div>
                    <Link name={createAccount} small={"true"} onClick={() => this.props.history.push("/signup")} />
                </div>
                <div>
                    <Button onClick={this.validate}>{signIn}</Button>
                </div>
            </div>
        </div>
    }


    //Layout JSX component
    Layout = (view) => {
        return <React.Fragment>
            <div className="login-container">
                <Card className="form">
                    {view}
                </Card>
                <LanguageSelection />
            </div>
        </React.Fragment>

    }

    render() {
        if (this.state.state === "login" && this.state.isSuggestionList)
            return this.Layout(this.LoginUserSuggestion())
        else if (this.state.state === "login")
            return this.Layout(this.Login())
        else if (this.state.state === "signUp")
            return this.Layout(this.SignUp())
        else
            return this.Layout(this.ForgetPassword())
    }

}


function mapDispatchToProps(dispatch) {
    return {
        addUser: (userDeatils) => dispatch(addUser(userDeatils))
    }
}


export default withRouter(connect(null, mapDispatchToProps)(Login))