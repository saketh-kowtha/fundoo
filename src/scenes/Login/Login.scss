@mixin span-focus {
	top: -9px;
	left: 12px;
	padding: 0px 5px;
	font-size: 14px;
}

@mixin active-warning-span {
	span {
		left: 20px;
		top: -8px;
		padding: 0px 5px;
	}
}

.login-container {
	display: flex;
	width: 100%;
	flex-flow: column;
	height: calc(100vh - 16px);
	justify-content: center;
	align-items: center;
	.form {
		padding: 50px 35px;
		min-height: 400px;
		.header {
			text-align: center;
			p {
				&:first-child {
					line-height: 10px;
					font-size: 30px;
                    @each $index, $color in (1: #5086ec, 2: #d85040, 3: #f0bc42, 4: #5086ec, 5: #77b57b, 6: #d85040) {        
						span:nth-child(#{$index}){
							color: $color
						}
					}
				}
			}
		}
		.footer {
			display: flex;
			margin-top: 30px;
			align-items: center;
			justify-content: space-between;
		}
		.input-group {
			position: relative;
			display: flex;
			flex-direction: column;
			margin: 0px 0px 28px 0px;
			.input-error-msg{
				margin-top: 60px; 
				font-size: 14px;
				color: #ff0000;
			}
			input {
				height: 50px;
				width: 350px;
				border-radius: 5px;
				padding: 0 20px;
				box-sizing: border-box;
				font-size: 18px;
				outline: none;
				border: 1px solid #e4e4e4;
				&:focus {
					border-color: #4688df;
					border-width: 2px;
					~ {
						span {
							@include span-focus();
							color: #4688df;
						}
					}
				}
			}
			span {
				position: absolute;
				top: 15px;
				left: 20px;
				transition: 0.2s;
				pointer-events: none;
				background: #fff;
				color: #808080;
			}
		}
		.active {
			input {
				border-color: #4688df;
				border-width: 2px;
				color: #000000;
			}
			span {
				@include span-focus();
				color: #4688df;
			}
		}
		.warning {
			input {
				border-color: #ff0000 !important;
				border-width: 2px;
				color: #ff0000 !important;
			}
			span {
				@include span-focus();
				color: #ff0000 !important;
			}
		}

		.readOnly{
			input{
				border-width: 0px 0px 1px 0px;
				border-radius: 0px;
				border-color: rgb(170, 170, 170) !important;
			}
			input:disabled {
				background-color : #ffff;
			}
		}
		.row {
			display: flex;
			flex-direction: row;
			margin: 0px 0px 10px 0px;
			input {
				font-size: 16px;
				width: 300px;
				margin-left: 10px;
				padding: 0 20px;
				&:focus {
					~ {
						span {
							left: 20px;
							top: -8px;
						}
					}
				}
			}
			.active, .warning {
				@include active-warning-span();
			}
		}
	}
	.login-users-list{
		width: 350px;
		li{
			border-bottom: 1px solid rgb(230, 230, 230);
			cursor: pointer;
			a{
				border-top-right-radius: 0px;
				border-bottom-right-radius: 0px;
			}
		}
	}
}
@media only screen and (max-width: 760px) {
	@mixin active-warning-span {
		span {
			left: 10px;
			font-size: 14px;
		}
	}
	.login-container {
		.form {
			box-shadow: none;
			padding: 50px 10px;
			.row {
				flex-direction: column;
				flex-wrap: wrap;
				input {
					margin-left: 0px;
					width: 350px;
					&:focus {
						~ {
							span {
								left: 10px;
								font-size: 14px;
							}
						}
					}
				}
				.active, .warning {
					@include active-warning-span();
				}
			}
		}
	}
}
