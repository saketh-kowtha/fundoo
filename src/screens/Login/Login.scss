

.login-container{
    display: flex;
    width: 100%;
    height: calc(100vh - 16px);
    justify-content: center;
    align-items: center;
    .form{
        height: fit-content;
        padding: 50px 35px;
        min-height: 400px;
        @media only screen and (max-width: 760px) {
            box-shadow: none;
            padding: 50px 10px;
        }
    
        @media only screen and (min-width: 760px) and (max-width: 1024px) {
            box-shadow: none;
            padding: 50px 10px;
        }
        .header{
            text-align: center;
            p:first-child {
                line-height: 10px;
                font-size : 30px;
                $logoColors: (1: rgb(80, 134, 236), 2: rgb(216, 80, 64), 3: rgb(240, 188, 66), 4: rgb(80, 134, 236), 5: rgb(119, 181, 123), 6: rgb(216, 80, 64));
                @each $index, $color in $logoColors {
                    span:nth-child(#{$index}) {
                    color: #{$color};
                    }
                }
            }
        }

        .footer{
            display: flex;
            margin-top: 30px;
            align-items: center;
            justify-content: space-between;
        }

        .input-group{
            position: relative;
            display: flex;
            flex-direction: column;
            margin: 0px 0px 28px 0px;
            input{
                height: 50px;
                width: 350px;
                border-radius: 5px;
                padding: 0 20px;
                box-sizing: border-box;
                font-size: 18px;
                outline: none;
                border: 1px solid #e4e4e4;
                &:focus{
                    border-color: #4688df;
                    border-width: 2px;
                }
                &:focus ~ span{
                    top: -9px;
                    left: 12px;
                    color: #4688df;
                    padding: 0px 5px;
                    font-size: 14px;
                }
            }

            span{
                position: absolute;
                top: 15px;
                left: 20px;
                transition: 0.2s;
                pointer-events: none;
                background: #fff;
                color: gray
            }
        }

        .active {
            input {
                border-color: #4688df;
                border-width: 2px;
                color: black;
            }

            span {
                top: -9px;
                left: 12px;
                color: #4688df;
                padding: 0px 5px;
                font-size: 14px;
            }

        }

        .warning {
            input {
                border-color: red !important;
                border-width: 2px;
            }
            span{
                top: -9px;
                left: 12px;
                padding: 0px 5px;
                font-size: 14px;
            }
            input,span {
                color: red !important;
            }
        }


        .row{
            display: flex;
            flex-direction: row;
            margin: 0px 0px 10px 0px;
            @media only screen and (max-width: 760px) {
                flex-direction: column;
                flex-wrap: wrap;
            }
        
            @media only screen and (min-width: 760px) and (max-width: 1024px) {
                flex-direction: column;
                flex-wrap: wrap;
            }
            input{
                font-size: 16px;
                width: 300px;
                margin-left: 10px;
                @media only screen and (max-width: 760px) {
                    margin-left: 0px;
                    width: 350px;
                }
            
                @media only screen and (min-width: 760px) and (max-width: 1024px) {
                    margin-left: 0px;
                    width: 350px;
                }
                padding: 0 20px;
                &:focus ~ span{
                    left: 20px;
                    top: -8px;
                    @media only screen and (max-width: 760px) {
                        left: 10px;
                    }
                
                    @media only screen and (min-width: 760px) and (max-width: 1024px) {
                        left: 10px;
                    }
                    font-size: 14px;
                }
            }

            .active span {
                top: -8px;
                left: 20px;
                @media only screen and (max-width: 760px) {
                    left: 10px;
                }
            
                @media only screen and (min-width: 760px) and (max-width: 1024px) {
                    left: 10px;
                }
                color: #4688df;
                padding: 0px 5px;
                font-size: 14px ;
            }    
    
            .warning span {
                top: -8px;
                left: 20px;
                @media only screen and (max-width: 760px) {
                    left: 10px;
                }
            
                @media only screen and (min-width: 760px) and (max-width: 1024px) {
                    left: 10px;
                }
                color: #4688df;
                padding: 0px 5px;
                font-size: 14px ;
            }    
        }

    }
}





