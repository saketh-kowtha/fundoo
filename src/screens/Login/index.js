import React from 'react'
import Card from '../../components/Card'

export default class Login extends React.Component {
    constructor() {
        super()
    }

    render() {
        return <div>
            {/* <Card elevated={true} >
                OK
            </Card> */}
        </div>
    }
}
