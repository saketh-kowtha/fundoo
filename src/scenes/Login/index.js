import React from 'react'
import {Card, Link, Button, LanguageSelection} from '../../components/'
import Input from './controllers/Input/Input'
import { 
    validateUsername, 
    validatePassword,
    validateEmail,
    validateLogin,
    validateSignUp
} from './controllers/Input/InputValidation'
import './Login.scss'
import geti18N from '../../strings'
const {
    title, 
    firstName, 
    lastName,
    email, 
    password, 
    confirmPassword, 
    signIn, 
    signUp, 
    signUpDesc, 
    signInDesc,
    forgotPasswd,
    createAccount,
    userName,
    send,
    forgetPasswordDesc,
    findYourMail
} = geti18N()
import http from '../../services/http'
import showToast from '../../components/Toast'
import { LOGIN, RESETPASSWORD, ERROR, SIGNUP } from '../../constants'
import { withRouter } from "react-router-dom";

/**
 * @name Login
 */
class Login extends React.Component{

    state={
        login: {
            username: {
                value: "",
                errMsg: ""
            },
            password: {
                value: "",
                errMsg: ""
            }
        },
        reset:{
            email: {
                value: "",
                errMsg: ""
            }
        },
        signUp: {
            firstname: {
                value: "Afs",
                errMsg: ""
            },
            lastname: {
                value: "Als",
                errMsg: ""
            },
            password: {
                value: "Admin@123",
                errMsg: ""
            },
            confirmpassword: {
                value: "Admin@123",
                errMsg: ""
            },
            email: {
                value: "aadarsh@gmail.com",
                errMsg: ""
            }
        },
        state: this.props.view || LOGIN
    }

    componentDidMount = () =>{
        this._isMount = true
    }

    componentWillUnmount = () =>{
        this._isMount = false
    }

    updateSate = (feild, value) => {
        if(!this._isMount)
            return
        if(!feild)
            this.setState({value})
        this.setState({feild: value})
    }

    handleUserInput = (name, input) =>{
        const stateData = {...this.state}
        stateData[this.state.state][name].value = input
        // this.setState({stateData})
        this.updateSate(null, stateData)
    }


    signupResponseHandler = (err, data) =>{
        if(err){
            const stateData = {...this.state.signUp}
            Object.keys(err).forEach(key => {
                let errMsg = (Array.isArray(err[key]) ? err[key].join(",") : err[key])
                stateData[key].errMsg = errMsg
                showToast(errMsg, ERROR)
            })
            // this.setState({[this.state.state]: stateData})
            this.updateSate([this.state.state], stateData)
        }
        else
            showToast(data)
    }

    resetResponseHandler = (err, data) => {
        if(err)
            showToast(err, ERROR)
        else
            showToast(data)
    }

    loginResponseHandler = (err, data) =>{
        if(err){
            showToast(err, ERROR)
        }
    }

    clearErrorMsgs = () => {
        let stateData = {...this.state[this.state.state]}
        Object.keys(stateData).forEach(feild => {
            stateData[feild].errMsg = ""
        })
        this.updateSate([this.state.state], stateData)
        // this.setState({
        //     [this.state.state] : stateData
        // })
    }

    validate =  () => {
        this.clearErrorMsgs()
        if(this.state.state === LOGIN)
        {
            const stateData = {...this.state.login}
            const loginValidation = validateLogin(this.state.login)
            if(loginValidation != true)
            {
                stateData[loginValidation.feild].errMsg = loginValidation.errorMsg
                // console.log(loginValidation, stateData[loginValidation.feild])
                this.updateSate("login", stateData)
                // this.setState({login: stateData})
            }
            else if(loginValidation === true)
            {
                http.login(stateData, this.loginResponseHandler)
            }
        }
        else if(this.state.state === RESETPASSWORD){
            const resetValidation = validateEmail(this.state.reset.email.value)
            const stateData = {...this.state.reset}
            if(resetValidation != true){
                stateData['email'].errMsg = resetValidation
                // this.setState({reset: stateData})    
                this.updateSate("reset", stateData)
            }
            else
                http.forgotPassword(stateData, this.resetResponseHandler)
        }
        else
        {
            const signUpValidation = validateSignUp(this.state.signUp)
            const stateData = {...this.state.signUp}
            if(signUpValidation != true)
            {
                stateData[signUpValidation.feild].errMsg = signUpValidation.errorMsg
                // this.setState({signUp: stateData})
                this.updateSate("signUp", stateData)

            }
            else if(signUpValidation === true)
            {
                http.signUp(stateData, this.signupResponseHandler)
            }
        }
    }

    ForgetPassword = () => {
        return <div>
                    <div className="header">
                        <p>{title.split("").map((char, index) => <span key={index}>{char}</span>)}</p>
                        <h2>{findYourMail}</h2>
                        <p>{forgetPasswordDesc}</p>
                    </div>
                    <div>
                        <Input 
                            name={"email"} 
                            key={email}
                            label={email} 
                            value={this.state.reset.email.value} 
                            errorMsg={this.state.reset.email.errMsg} 
                            onChange={this.handleUserInput} 
                            validate={validateEmail}/>
                    </div>
                    <div className="footer">
                        <div>
                            <Link 
                                name={signIn} 
                                small={"true"} 
                                onClick={() => this.props.history.push("/login")}/>
                        </div>
                        <div>
                            <Button onClick={this.validate}>{send}</Button>
                        </div>
                    </div>
                </div>
    }


    SignUp = () => {
        let _this = {...this.state.signUp}
        return  <div>
                    <div className="header">
                        <p>{title.split("").map((char, index) => <span key={index}>{char}</span>)}</p>
                        <h2>{signUp}</h2>
                        <p>{signUpDesc}</p>
                    </div>
                    <div className="row">
                        <Input 
                            name={"firstname"} 
                            key={firstName} 
                            label={firstName} 
                            value={_this.firstname.value} 
                            errorMsg={_this.firstname.errMsg} 
                            onChange={this.handleUserInput} 
                            validate={validateUsername}/>

                        <Input 
                            name={"lastname"} 
                            key={lastName} 
                            label={lastName} 
                            value={_this.lastname.value} 
                            errorMsg={_this.lastname.errMsg} 
                            onChange={this.handleUserInput}  
                            validate={validateUsername}/>
                    </div>
                    <div className="row" >
                        <Input 
                            name={"email"} 
                            key={email} 
                            label={email} 
                            value={_this.email.value} 
                            errorMsg={_this.email.errMsg} 
                            onChange={this.handleUserInput} 
                            validate={validateEmail}/>
                    </div>
                    <div className="row">
                        <Input 
                            name={"password"} 
                            key={password} 
                            label={password} 
                            value={_this.password.value} 
                            errorMsg={_this.password.errMsg} 
                            secret={true} 
                            onChange={this.handleUserInput} 
                            validate={validatePassword}/>
                        <Input 
                            name={"confirmPassword"} 
                            key={confirmPassword}  
                            label={confirmPassword}  
                            value={_this.confirmpassword.value} 
                            errorMsg={_this.confirmpassword.errMsg} 
                            onChange={this.handleUserInput} 
                            secret={true} 
                            validate={validatePassword}/>
                    </div>
                    <div className="footer">
                        <div>
                            <Link 
                                name={signIn} 
                                small={"true"} 
                                onClick={() => this.props.history.push("/login")}/>
                        </div>
                        <div>
                            <Button 
                                onClick={this.validate}>
                                    {signUp}
                            </Button>
                        </div>
                    </div>
                </div>
    }

    Login = () => {
        return  <div>
                    <div className="header">
                        <p>{title.split("").map((char, index) => <span key={index}>{char}</span>)}</p>
                        <h2>{signIn}</h2>
                        <p>{signInDesc} </p>
                    </div>
                    <div>
                        <Input 
                            name={"username"} 
                            key={userName}
                            label={userName}
                            errorMsg={this.state.login.username.errMsg} 
                            onChange={this.handleUserInput} 
                            validate={validateEmail}/>
                        <Input 
                            name={"password"} 
                            key={password}
                            label={password}
                            errorMsg={this.state.login.password.errMsg} 
                            onChange={this.handleUserInput} 
                            secret={true} 
                            validate={validatePassword}/>
                    </div>
                    <div>                                
                        <Link name={forgotPasswd} small={"true"} onClick={() => this.props.history.push("/resetpassword")}/>
                    </div>
                    <div className="footer">
                        <div>
                            <Link name={createAccount} small={"true"} onClick={() => this.props.history.push("/signup")}/>
                        </div>
                        <div>
                            <Button onClick={this.validate}>{signIn}</Button>
                        </div>
                    </div>
                </div>
    }
    

    Layout = (view) => {
        return <React.Fragment>
                    <div className="login-container">
                        <Card className="form">
                            {view}
                        </Card>
                        <LanguageSelection />
                    </div>
                </React.Fragment>

    }

    render(){
        if(this.state.state === "login")
            return this.Layout(this.Login())
        else if(this.state.state === "signUp")
            return this.Layout(this.SignUp())
        else
            return this.Layout(this.ForgetPassword())
    }

}



export default withRouter(Login)