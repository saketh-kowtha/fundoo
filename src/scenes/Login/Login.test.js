import React from 'react';
import { shallow, mount } from 'enzyme';
import Login from './Login';


const sleep = m => new Promise(r => setTimeout(r, m))

describe('Login', () => {
    const component = mount(<Login view="login" />);


    it("Should Show error message on Invalid Email", () => {
        component.find("input[name='username']").simulate('change', {target: {value: "abc", name: "username"}})
        component.find("button").simulate('click')
        expect(component.find("input[name='username']~span~span").text()).toBe("Must contain @ symbol")
    })

    it("Should Show error message on Empty Email", () => {
        component.find("input[name='username']").simulate('change', {target: {value: "", name: "username"}})
        component.find("button").simulate('click')
        expect(component.find("input[name='username']~span~span").text()).toBe("Must contain @ symbol")
    })

    it("Should Show error message on Empty Password", () => {
        component.find("input[name='username']").simulate('change', { target: { value: "abc@gmail.com", name: "username" } })
        component.find("input[name='password']").simulate('change', {target: {value: "", name: "password"}})
        component.find("button").simulate('click')
        expect(component.find("input[name='password']~span~span").text()).toBe("Password Should not be empty")
    })

    it("Should Show error message on Invalid Credintails", async () => {
        component.find("input[name='username']").simulate('change', { target: { value: "abc@gmail.com", name: "username" } })
        component.find("input[name='password']").simulate('change', {target: {value: "Amin@123", name: "password"}})
        component.find("button").simulate('click')
        await sleep(500)
        expect(document.querySelectorAll("div[class='toast error']")[0].innerHTML).toBe("Invalid Credintails")
    })

    it("Should match snapshot", async () => {
        component.find("input[name='username']").simulate('change', { target: { value: "aadarsh@gmail.com", name: "username" } })
        component.find("input[name='password']").simulate('change', {target: {value: "Admin@123", name: "password"}})
        component.find("button").simulate('click')
        await sleep(500)
        expect(component).toMatchSnapshot()
    })


});





describe('Signup', () => {
    const component = mount(<Login view="signUp" />);


    it("Should Show error message on Invalid Firstname", () => {
        component.find("input[name='firstname']").simulate('change', {target: {value: "", name: "firstname"}})
        component.find("button").simulate('click')
        expect(component.find("input[name='firstname']~span~span").text()).toBe("Username Should not be empty")
    })

    it("Should Show error message on Invalid Lastname", () => {
        component.find("input[name='firstname']").simulate('change', {target: {value: "abc", name: "firstname"}})
        component.find("input[name='lastname']").simulate('change', {target: {value: "", name: "lastname"}})
        component.find("button").simulate('click')
        expect(component.find("input[name='lastname']~span~span").text()).toBe("Username Should not be empty")
    })

    // it("Should Show error message on Invalid Email", () => {
    //     component.find("input[name='firstname']").simulate('change', {target: {value: "abc", name: "firstname"}})
    //     component.find("input[name='lastname']").simulate('change', {target: {value: "xyz", name: "lastname"}})
    //     component.find("input[name='email']").simulate('change', {target: {value: "", name: "email"}})
    //     component.find("button").simulate('click')
    //     expect(component.find("input[name='email']~span~span").text()).toBe("Must contain @ symbol")
    // })

});