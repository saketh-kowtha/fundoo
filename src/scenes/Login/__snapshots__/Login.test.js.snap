// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Should Show error message on Invalid Credintails 1`] = `ReactWrapper {}`;

exports[`Login Should be snapshot 1`] = `ReactWrapper {}`;

exports[`Login Should match snapshot 1`] = `ReactWrapper {}`;
